frappe/core/doctype/user/user.py

```diff
@@ -44,6 +44,29 @@ def before_insert(self):
 	def after_insert(self):
 		self.set_default_roles()
 
+	def force_user_email_update(self):
+		for user_email in self.user_emails:
+			if not user_email.email_id:
+				user_email.email_id = frappe.db.get_value("Email Account", {"name": user_email.email_account},
+				                                          "email_id")
+
+	def user_emails_to_permissions(self):
+		if frappe.session.user == "Administrator" or "System Manager" in frappe.get_roles():
```

Comment:
You want to have the ability to assign accounts for each users. As in many-to-many?

I am not sure of the use case myself. Even Google Apps does not allow it :) You can make someone the email admin and then that user has the access or not.

Maybe a check to state whether this user has access to all accounts or not should do?